rbc/external.py

```diff
@@ -2,69 +2,96 @@
 """
 
 # This code has heavily inspired in the numba.extending.intrisic code
+import warnings
 from typing import Dict, List, Union
 from collections import defaultdict
 from numba.core import extending, funcdesc, types, typing
 from rbc.typesystem import Type
 from rbc.targetinfo import TargetInfo
+from rbc.errors import UnsupportedError
+from rbc.utils import validate_devices
 
 
 class External:
+
     @classmethod
-    def external(cls, *args):
-        """
-        Parameters
+    def external(cls, *args, **kwargs):
```

Comment:
Given that all use cases we have for this function is a single argument, what about replacing it by:

```suggestion
    def external(cls, signature, *, devices=["CPU"]):
```

Reply:
Done. This also simplified external implementation.